### DISCUSSION TITLE
Create and append interface dynamically to an object

### DISCUSSION THREAD

**Discussion by Krulle86** [Kistler-Group/sdbus-cpp#417] [General]
Hey,

is there a way to create and append dynamically interfaces to an object, without to destroy and recreate the whole object?

For example:
**Object** -> _/org/sdbuscpp/Concatenator_
after a certain period
**Create and append** -> _org.sdbuscpp.Concatenator1_
**Create and append** -> _org.sdbuscpp.Concatenator2_
**Create and append** -> _org.sdbuscpp.Concatenator3_
.
.
.
Any ideas?

Thanks!

**Comment by sangelovic** [comment]
@Krulle86 It's not possible in `v1.x` releases, but it has become possible in the upcoming `v2.0` release (#364) -- there is a new `addVTable()` method and its overload that returns a slot object. With that, one can dynamically add or remove object interfaces at run time. See the tip in https://github.com/Kistler-Group/sdbus-cpp/blob/release/v2.0/docs/using-sdbus-c%2B%2B.md#server-side-1
